File: src/android/src/main/kotlin/com/circuids/boat/pipeline/PipelineConfig.kt
package com.circuids.boat.pipeline

/**
 * Immutable pipeline configuration captured at construction.
 */
data class PipelineConfig(
    val frameDurationMs: Int = 20,
    val deadlineFraction: Double = 0.80,
    val vadEnabled: Boolean = false,
    val diagnosticsEnabled: Boolean = false,
) {
    companion object {
        fun fromMap(map: Map<String, Any?>): PipelineConfig = PipelineConfig(
            frameDurationMs = (map["bufferDurationMs"] as? Number)?.toInt() ?: 20,
            deadlineFraction = (map["deadlineFraction"] as? Number)?.toDouble() ?: 0.80,
            vadEnabled = map["vadEnabled"] as? Boolean ?: false,
            diagnosticsEnabled = map["diagnosticsEnabled"] as? Boolean ?: false,
        )
    }
}

File: src/ios/Classes/Pipeline/PipelineConfig.swift
import Foundation

/// Immutable pipeline configuration captured at construction.
struct PipelineConfig {
    let frameDurationMs: Int
    let deadlineFraction: Double
    let vadEnabled: Bool
    let diagnosticsEnabled: Bool

    init(frameDurationMs: Int = 20, deadlineFraction: Double = 0.80,
         vadEnabled: Bool = false, diagnosticsEnabled: Bool = false) {
        self.frameDurationMs = frameDurationMs
        self.deadlineFraction = deadlineFraction
        self.vadEnabled = vadEnabled
        self.diagnosticsEnabled = diagnosticsEnabled
    }

    static func fromMap(_ map: [String: Any]) -> PipelineConfig {
        PipelineConfig(
            frameDurationMs: map["bufferDurationMs"] as? Int ?? 20,
            deadlineFraction: (map["deadlineFraction"] as? Double) ?? 0.80,
            vadEnabled: map["vadEnabled"] as? Bool ?? false,
            diagnosticsEnabled: map["diagnosticsEnabled"] as? Bool ?? false
        )
    }
}
